### PR TITLE
Show import error for 'airflow dags list' CLI command

### PR DESCRIPTION
**Pull request by ephraimbuddy**
When there's an import error in a dag, the dag doesn't show up in the list of dags
and there's no indication that there's an import error.
This PR fixes that

### CODE REVIEW

File: airflow/cli/commands/dag_command.py [hunk 1]
```diff
@@ -315,6 +315,12 @@ def dag_next_execution(args):
 def dag_list_dags(args):
     """Displays dags with or without stats at the command line"""
     dagbag = DagBag(process_subdir(args.subdir))
+    if dagbag.import_errors:
+        from rich import print as rich_print
+
+        rich_print("[red][bold]Error:[bold] The following dags failed to load [red]")
```
Comment (jedcunningham) [inline airflow/cli/commands/dag_command.py]:
```suggestion
        rich_print("[red][bold]Error:[/bold] The following dags failed to load")
```

File: airflow/cli/commands/dag_command.py [hunk 2]
```diff
@@ -315,6 +315,12 @@ def dag_next_execution(args):
...
+        for filename, stacktrace in dagbag.import_errors.items():
+            rich_print(f"[red][bold]Broken DAG: {filename}: {stacktrace}[bold][red]")
```
Comment (jedcunningham) [inline airflow/cli/commands/dag_command.py]:
```suggestion
            rich_print(f"[red][bold]Broken DAG {filename}:[/bold] {stacktrace}")
```

Not sure we even need to bold the filename?

**Review by jedcunningham — CHANGES_REQUESTED** [review]

**Comment by mik-laj** [comment]
Won't that cause problem for scripts that process the output of this command? I think we should display import errors on stderr.

**Comment by mik-laj** [comment]
Related PR: https://github.com/apache/airflow/pull/15087

**Comment by ephraimbuddy** [comment]
> Related PR: #15087

What's your opinion about having `--show-import-error` option? I think printing the error in stderr without any option is OK.
Happy to hear what you think then I would work on it and add tests

**Comment by mik-laj** [comment]
@ephraimbuddy, For me, displaying an error on stderr without any options is also OK. In the above-mentioned PR, I proposed to add a separate command as this would allow us to write scripts that process these import errors, but we can do it as a separate contribution.

**Comment by ephraimbuddy** [comment]
@mik-laj I have created a separate command for import errors(PR: https://github.com/apache/airflow/pull/22084). Would wait for that to be merged and update this PR to just say that we have import errors and ask the users to run `airflow dags list-import-errors` to see the errors. WDYT?

File: tests/cli/commands/test_dag_command.py [hunk 3]
```diff
@@ -407,6 +407,15 @@ def test_cli_list_dags(self):
         assert "airflow/example_dags/example_complex.py" in out
         assert "- dag_id:" in out
 
+    @conf_vars({('core', 'load_examples'): 'false'})
+    def test_cli_list_dags_prints_import_errors(self):
+        dag_path = os.path.join(TEST_DAGS_FOLDER, 'test_invalid_cron.py')
+        args = self.parser.parse_args(['dags', 'list', '--output', 'yaml', '--subdir', dag_path])
+        with contextlib.redirect_stderr(io.StringIO()) as temp_stdout:
+            dag_command.dag_list_dags(args)
+            out = temp_stdout.getvalue()
```
Comment (ephraimbuddy) [inline tests/cli/commands/test_dag_command.py]:
```suggestion
        with contextlib.redirect_stderr(io.StringIO()) as temp_stderr:
            dag_command.dag_list_dags(args)
            out = temp_stderr.getvalue()
```

**Review by mik-laj — APPROVED** [review]

File: airflow/cli/commands/dag_command.py [hunk 4]
```diff
@@ -315,6 +315,14 @@ def dag_next_execution(args):
 def dag_list_dags(args):
     """Displays dags with or without stats at the command line"""
     dagbag = DagBag(process_subdir(args.subdir))
+    if dagbag.import_errors:
+        from rich import print as rich_print
+
+        rich_print(
+            "[red][bold]Error:[/bold] Failed to load all files. "
+            "For details, run airflow dags list-import-errors",
```
Comment (jedcunningham) [inline airflow/cli/commands/dag_command.py]:
```suggestion
            "For details, run `airflow dags list-import-errors`",
```
nit

**Review by jedcunningham — APPROVED** [review]

**Review by potiuk — APPROVED** [review]

**Pull request merged**